internal/pkg/otel/configtranslate/otelconfig_test.go

```diff
@@ -0,0 +1,335 @@
+								Type:   client.UnitTypeInput,
+								Config: component.MustExpectedConfig(fileStreamConfig),
+							},
+							{
+								ID:     "test-default",
+								Type:   client.UnitTypeOutput,
+								Config: component.MustExpectedConfig(esOutputConfig),
+							},
+						},
+					},
+				},
+			},
+			expectedConfig: confmap.NewFromStringMap(map[string]any{
+				"exporters": map[string]any{
+					"elasticsearch/_agent-component/default": map[string]any{
+						"batcher": map[string]any{
+							"enabled":        true,
+							"max_size_items": 1600,
+						},
+						"mapping": map[string]any{
+							"mode": "bodymap",
+						},
+						"endpoints": []string{"http://localhost:9200"},
+						"password":  "password",
+						"user":      "elastic",
+						"retry": map[string]any{
+							"enabled":          true,
+							"initial_interval": 1 * time.Second,
+							"max_interval":     1 * time.Minute,
+							"max_retries":      3,
+						},
+						"logs_dynamic_index": map[string]any{
+							"enabled": true,
+						},
+						"num_workers":       0,
+						"api_key":           "",
+						"logs_index":        "filebeat-9.0.0",
+						"timeout":           90 * time.Second,
+						"idle_conn_timeout": 3 * time.Second,
+						"metrics_dynamic_index": map[string]any{
+							"enabled": true,
+						},
+					},
+				},
+				"receivers": map[string]any{
+					"filebeatreceiver/_agent-component/test": map[string]any{
+						"filebeat": map[string]any{
+							"inputs": []map[string]any{
+								{
+									"id":         "test",
+									"type":       "filestream",
+									"paths":      map[string]any{},
+									"index":      "logs-generic-default",
+									"processors": defaultProcessors,
+								},
+							},
+						},
+						"output": map[string]any{
+							"otelconsumer": map[string]any{},
+						},
+						"path": map[string]any{
+							"data": filepath.Join(paths.Home(), "run", "test"),
+						},
+					},
+				},
+				"service": map[string]any{
+					"pipelines": map[string]any{
+						"logs/_agent-component/test": map[string][]string{
+							"exporters": []string{"elasticsearch/_agent-component/default"},
+							"receivers": []string{"filebeatreceiver/_agent-component/test"},
+						},
+					},
+				},
+			}),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualConf, actualError := GetOtelConfig(tt.model, agentInfo)
+			if actualConf == nil || tt.expectedConfig == nil {
+				assert.Equal(t, tt.expectedConfig, actualConf)
+			} else { // this gives a nicer diff
+				assert.Equal(t, tt.expectedConfig.ToStringMap(), actualConf.ToStringMap())
+			}
+
+			if actualConf != nil {
+				t.Logf("%v", actualConf.ToStringMap())
+			}
+			if tt.expectedError != nil {
+				assert.Error(t, actualError)
+				assert.EqualError(t, actualError, tt.expectedError.Error())
+			} else {
+				assert.NoError(t, actualError)
+			}
+		})
+	}
+}
+
+// TODO: Add unit tests for other config generation functions
```

Comment:
What other tests are missing here? Are they going to be added in this or in a follow-up PR ?